src/main/java/seedu/address/model/residence/Residence.java

```diff
@@ -106,6 +107,16 @@ public boolean isSameResidence(Residence otherResidence) {
                 && otherResidence.getResidenceName().equals(getResidenceName());
     }
 
+    /**
+     * Returns true if this residence has any bookings that start in the next 7 days.
+     */
+    public boolean hasUpcomingBooking() {
+        return this.bookingList.getValue().stream()
+                .filter(b -> b.getStart().isAfter(LocalDate.now())
```

Comment:
Just to clarify, should upcoming bookings include bookings on the current date and bookings 7 days from the current date? The use of isAfter and isBefore will return false for boundary values.